src/data/set6/champions.ts

```diff
@@ -53,6 +53,22 @@ export default {
 		},
 	},
 
+	[ChampionKey.Kassadin]: {
+		cast: (elapsedMS, spell, champion) => {
+			const target = champion.target
+			const manaReave = champion.getSpellVariable(SpellKey.ManaReave) ?? 0
+			const castTime = champion.getCurrentSpell()?.castTime ?? 0
+			if (!target) { return console.log('No target', champion.name, champion.team) }
+			champion.queueProjectile(elapsedMS, spell, {
+				onCollision: () => {
+					if (!target.getBonusesFrom(SpellKey.ManaReave).length) {
+						target.bonuses.push([SpellKey.ManaReave, [[BonusKey.ManaReductionPercent, -100 * manaReave, elapsedMS + castTime]]]) 
```

Comment:
This change will expire the `ManaReductionPercent` effect on the _target_ after the `castTime` ends. Which is not quite right. Note that we want the _source_ to gain the damage reduction (Kassadin becomes tankier after his spell reaches his target).

Reply:
This is pseudocode, but something like:
```ts
target.addBonuses(SpellKey.ManaReave, [BonusKey.ManaReductionPercent, -100 * manaReave])
champion.setBonusesFor(ChampionKey.Kassadin, [BonusKey.DamageReduction, champion.getSpellVariable(SpellKey.DamageReduction) / 100, champion.getSpellVariable('Duration') * 1000])
```
Note that `addBonuses` creates new entries while `setBonusesFor` _replaces_ any existing one. I'm not sure but I believe damage reduction from his spell does not stack on himself.